src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -57,16 +66,28 @@ struct LocationStructureWrapper : public chip::app::Clusters::ServiceArea::Struc
      *
      * @note Requirements regarding what combinations of fields and values are valid are not checked by this class.
      * @note If aLocationName is larger than kLocationNameMaxSize, it will be truncated.
-     * @note If aLocationName is an empty string and aFloorNumber and aAreaTypeTag are null, locationInfo will be set to null.
+     * @note If aLocationName is an empty string and aFloorNumber and aAreaType are null, locationInfo.locationinfo will be set to null.
      */
-    LocationStructureWrapper(uint32_t aLocationId, const DataModel::Nullable<uint8_t> & aMapId, const CharSpan & aLocationName,
+    LocationStructureWrapper(uint32_t aLocationId, 
```

Comment:
note that to actually fix #34519 this constructor should not exist.

Reply:
It is not readable or maintainable to have methods with more than 3-4 arguments. This has 8 .. at some point we will add the 9th..this is not practical.

Reply:
What do you suggest for the constructor?

Reply:
Builder-style is very easy to use, with a good default constructor.

Then you can have 

```
LocationStructureWrapper locationStructure;

locationStructure.SetXxx(xxx).SetYyy(yyy).
```

This can also be used in-place:

```
DoSomething(LocationStructureWrapper{}.SetXxx().SetYyy(yyy));
```

Setters must then simply return `LocationStructureWrapper &` instead of `void`.

Reply:
updated